Source/WebKit/UIProcess/mac/DisplayLink.h

```diff
@@ -27,70 +27,71 @@
 
 #if HAVE(CVDISPLAYLINK)
 
-#include "Connection.h"
 #include "DisplayLinkObserverID.h"
 #include <CoreVideo/CVDisplayLink.h>
 #include <WebCore/AnimationFrameRate.h>
 #include <WebCore/DisplayUpdate.h>
 #include <WebCore/PlatformScreen.h>
-#include <wtf/HashMap.h>
+#include <wtf/HashSet.h>
 #include <wtf/Lock.h>
 
-namespace IPC {
-class Connection;
-}
-
 namespace WebKit {
 
 class DisplayLink {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    explicit DisplayLink(WebCore::PlatformDisplayID);
-    ~DisplayLink();
-    
-    void addObserver(IPC::Connection::UniqueID, DisplayLinkObserverID, WebCore::FramesPerSecond);
-    void removeObserver(IPC::Connection::UniqueID, DisplayLinkObserverID);
-    void removeObservers(IPC::Connection::UniqueID);
-
-    void incrementFullSpeedRequestClientCount(IPC::Connection::UniqueID);
-    void decrementFullSpeedRequestClientCount(IPC::Connection::UniqueID);
+    class Client {
+    friend class DisplayLink;
+    public:
+        virtual ~Client() = default;
+        
+    private:
+        virtual void displayLinkFired(WebCore::PlatformDisplayID, WebCore::DisplayUpdate, bool wantsFullSpeedUpdates, bool anyObserverWantsCallback) = 0;
+    };
 
-    void setPreferredFramesPerSecond(IPC::Connection::UniqueID, DisplayLinkObserverID, WebCore::FramesPerSecond);
+    DisplayLink(WebCore::PlatformDisplayID);
+    ~DisplayLink();
 
     WebCore::PlatformDisplayID displayID() const { return m_displayID; }
-    
     WebCore::FramesPerSecond nominalFramesPerSecond() const { return m_displayNominalFramesPerSecond; }
 
-    // When responsiveness is critical, we send the IPC to a background queue. Otherwise, we send it to the
-    // main thread to avoid unnecessary thread hopping and save power.
-    static void setShouldSendIPCOnBackgroundQueue(bool value) { shouldSendIPCOnBackgroundQueue = value; }
+    void addClient(Client&);
+    void removeClient(Client&);
```

Comment:
Where are these functions implemented? I do not see they are ever called.

Reply:
`removeClient()` is called on web process teardown. `addclient()` was never implemented; removed.